src/NUnitFramework/framework/Internal/Results/TestResult.cs

```diff
@@ -80,7 +80,13 @@ public abstract class TestResult : ITestResult
         private string _message;
         private string _stackTrace;
 
-        private List<AssertionResult> _assertionResults = new List<AssertionResult>();
+        private readonly List<AssertionResult> _assertionResults = new List<AssertionResult>();
+
+        /// <summary>
+        /// TestAttachments attached to result.
+        /// Only exposed for testing.
+        /// </summary>
+        internal ICollection<TestAttachment> TestAttachments { get; } = new List<TestAttachment>();
```

Comment:
Working my way down... I can't see why this would not be `private readonly List`. Maybe it will become clear as I go. 😄